com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs

```diff
@@ -141,76 +400,150 @@ public SceneSwitchProgress SwitchScene(string sceneName, LoadSceneMode loadScene
-            };
+            if (!m_ScenesLoaded.ContainsKey(sceneName) || !m_ScenesLoaded[sceneName].ContainsKey(scene.handle))
+            {
+                Debug.LogError($"{nameof(UnloadScene)} internal error! {sceneName} with handle {scene.handle} is not within the internal scenes loaded dictionary!");
+                return SceneEventProgressStatus.InternalNetcodeError;
+            }
 
-            // Move ALL NetworkObjects to the temp scene
-            MoveObjectsToDontDestroyOnLoad();
+            SceneEventData.SceneEventGuid = sceneEventProgress.Guid;
+            SceneEventData.SceneEventType = SceneEventData.SceneEventTypes.S2C_Unload;
+            SceneEventData.SceneIndex = SceneNameToIndex[sceneName];
+            SceneEventData.SceneHandle = m_ScenesLoaded[sceneName][scene.handle].handle;
 
-            IsSpawnedObjectsPendingInDontDestroyOnLoad = true;
+            // This will be the message we send to everyone when this scene event sceneEventProgress is complete
+            sceneEventProgress.SceneEventType = SceneEventData.SceneEventTypes.S2C_UnLoadComplete;
 
-            // Switch scene
-            AsyncOperation sceneLoad = SceneManager.LoadSceneAsync(sceneName, loadSceneMode);
+            // Sends the unload scene notification
+            SendSceneEventData(m_NetworkManager.ConnectedClientsIds.Where(c => c != m_NetworkManager.ServerClientId).ToArray());
 
-            s_NextSceneName = sceneName;
+            m_ScenesLoaded[sceneName].Remove(scene.handle);
 
-            sceneLoad.completed += (AsyncOperation asyncOp2) => { OnSceneLoaded(switchSceneProgress.Guid, null); };
-            switchSceneProgress.SetSceneLoadOperation(sceneLoad);
-            OnSceneSwitchStarted?.Invoke(sceneLoad);
+            if (m_ScenesLoaded[sceneName].Count == 0)
+            {
+                m_ScenesLoaded.Remove(sceneName);
+            }
 
-            return switchSceneProgress;
+            AsyncOperation sceneUnload = SceneManager.UnloadSceneAsync(scene);
+            sceneUnload.completed += (AsyncOperation asyncOp2) => { OnSceneUnloaded(); };
+            sceneEventProgress.SetSceneLoadOperation(sceneUnload);
+
+            // Notify local server that a scene is going to be unloaded
+            OnSceneEvent?.Invoke(new SceneEvent()
+            {
+                AsyncOperation = sceneUnload,
+                SceneEventType = SceneEventData.SceneEventType,
+                LoadSceneMode = SceneEventData.LoadSceneMode,
+                SceneName = sceneName,
+                ClientId = m_NetworkManager.ServerClientId  // Server can only invoke this
+            });
+
+            //Return the status
+            return sceneEventProgress.Status;
         }
 
-        // Called on client
-        internal void OnSceneSwitch(uint sceneIndex, Guid switchSceneGuid, Stream objectStream)
+        /// <summary>
+        /// Client Side:
+        /// SceneManager.UnloadSceneAsync handler for clients
+        /// </summary>
```

Comment:
nit: afaics, this is the handler for SceneEventTypes.S2C_Unload, not a callback called directly by SceneManager.UnloadSceneAsync